targets/bmv2/fuzzer.cpp

```diff
@@ -9,7 +11,102 @@ const Bmv2V1ModelProgramInfo &Bmv2V1ModelFuzzer::getProgramInfo() const {
     return *P4RuntimeFuzzer::getProgramInfo().checkedTo<Bmv2V1ModelProgramInfo>();
 }
 
-InitialP4RuntimeConfig Bmv2V1ModelFuzzer::produceInitialConfig() { return {}; }
+void addTernary(p4::v1::TableEntry *protoEntry, int fieldId, int bitwidth) {
+    auto value = Utils::getRandConstantForWidth(bitwidth)->value;
+    std::optional<std::string> valueStr = P4::ControlPlaneAPI::stringReprConstant(value, bitwidth);
+    std::optional<std::string> maskStr = P4::ControlPlaneAPI::stringReprConstant(0, bitwidth);
+
+    auto protoMatch = protoEntry->add_match();
+    protoMatch->set_field_id(fieldId);
+    auto protoTernary = protoMatch->mutable_ternary();
+    protoTernary->set_value(*valueStr);
+    protoTernary->set_mask(*maskStr);
+}
+
+void addActions(p4::v1::TableEntry *tableEntry, const p4::config::v1::Table *table, google::protobuf::RepeatedPtrField<p4::config::v1::Action> *actions) {
+    const auto &action_refs = table->action_refs();
+    auto action_index = Utils::getRandInt(action_refs.size()-1);
+
+    auto protoAction = tableEntry->mutable_action()->mutable_action();
+    auto action_ref_id = action_refs[action_index].id();
+    protoAction->set_action_id(action_ref_id);
+
+    for (auto &action : *actions) {
```

Comment:
Do not iterate over the actions, instead pick one that is available.

Reply:
Hm I would split this code up, I would also make use of https://github.com/p4lang/p4c/blob/main/backends/p4tools/common/control_plane/p4info_map.h and https://github.com/p4lang/p4c/blob/main/control-plane/p4infoApi.h